core/src/main/java/org/commonjava/maven/ext/manip/impl/InfoManipulator.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (C) 2012 Red Hat, Inc. (jcasey@redhat.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.commonjava.maven.ext.manip.impl;
+
+import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.component.annotations.Requirement;
+import org.commonjava.maven.ext.manip.ManipulationException;
+import org.commonjava.maven.ext.manip.ManipulationManager;
+import org.commonjava.maven.ext.manip.ManipulationSession;
+import org.commonjava.maven.ext.manip.io.PomIO;
+import org.commonjava.maven.ext.manip.model.Project;
+import org.commonjava.maven.ext.manip.state.InfoState;
+import org.commonjava.maven.ext.manip.state.InfoState.GAV;
+
+import java.util.List;
+
+/**
+ * {@link Manipulator} implementation that gathers various information about the project.
+ * The information is intended to be saved into {@link ManipulationManager#RESULT_FILE}
+ */
+@Component( role = Manipulator.class, hint = "info-manipulator" ) public class InfoManipulator
+                extends AbstractNoopManipulator
+{
+    @Requirement private PomIO pomIO;
+
+    @Override public void init( final ManipulationSession session )
+    {
+        session.setState( new InfoState() );
+    }
+
+    private Project getExecutionRoot( final List<Project> projects )
+    {
+        for ( Project project : projects )
+        {
+            if ( project.isExecutionRoot() )
+            {
+                return project;
+            }
+        }
+        throw new IllegalStateException( "Execution root not found in " + projects + "." );
+    }
+
+    @Override public void afterApplyChanges( List<Project> projects, ManipulationSession session )
+                    throws ManipulationException
+    {
+        Project executionRoot = getExecutionRoot( projects );
+        // Have to do this again to get modified values:/
+        final List<Project> projectsModified = pomIO.parseProject( executionRoot.getPom() );
+        executionRoot = getExecutionRoot( projectsModified );
+
```

Comment:
I don't think this is required ; the Project instance contains the Model undergoing modification. Therefore that already has the correct GAV. 
Do we even need a new manipulator - the VersioningManipulator could fill in the InfoState once its completed altering the project?